src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -77,10 +77,45 @@ public ElementNode(AbstractNode root, AbstractNode parent, DetailAST detailAst)
         this.detailAst = detailAst;
         text = TokenUtil.getTokenName(detailAst.getType());
         indexAmongSiblings = parent.getChildren().size();
+        setDepth(parent.getDepth() + 1);
         createTextAttribute();
         createChildren();
     }
 
+    /**
+     * Compares current object with specified for order.
+     * @param other another {@code NodeInfo} object
+     * @return number representing order of current object to specified one
+     */
+    @Override
+    public int compareOrder(NodeInfo other) {
+        int result = 0;
+        if (other instanceof ElementNode) {
+            result = getDepth() - ((AbstractNode) other).getDepth();
```

Comment:
We check instance to be ElementNode but do a cast to AbstractNode. Shouldn't both be the same class?

Reply:
@rnveach Yes, you are right, I forgot to change after a few changes in other code.
I've corrected it.

Reply:
@gaurabdg It looks like IntelliJ wants both to be `AbstractNode`. I don't see any parts of the code requiring it be `ElementNode`.
https://teamcity.jetbrains.com/viewLog.html?buildId=2866738&tab=Inspection&buildTypeId=Checkstyle_IdeaInspectionsPullRequest

Reply:
Done.